### ISSUE TITLE
ResolvePackageNotFound

### ISSUE THREAD

**Issue by Elahe96** [xl-sr/CAL#26]
### conda env create -f requirements.yml

Collecting package metadata (repodata.json): done
Solving environment: failed

ResolvePackageNotFound:
  - zlib==1.2.11=ha838bed_2
  - tk==8.6.8=hbc83047_0
  - tensorflow-base==1.10.0=gpu_py27had579c0_0
  - bcolz==1.2.1=py27h04863e7_0
  - readline==7.0=h7b6447c_5
  - yaml==0.1.7=had09818_2
  - protobuf==3.6.0=py27hf484d3e_0
  - jpeg==8d=2
  - blosc==1.14.4=hdbcaa40_0
  - numpy-base==1.15.1=py27h81de0dd_0
  - singledispatch==3.4.0.3=py27_0
  - locket==0.2.0=py27_1
  - lcms==1.19=0
  - cudatoolkit==9.2=0
  - hdf5==1.10.2=hba1933b_1
  - partd==0.3.8=py27_0
  - h5py==2.8.0=py27h989c5e5_3
  - pyyaml==3.13=py27h14c3975_0
  - mkl_random==1.0.1=py27h4414c95_1
  - numpy==1.15.1=py27h1d66e8a_0
  - markupsafe==1.0=py27h14c3975_1
  - ncurses==6.1=hf484d3e_0
  - tblib==1.3.2=py27_0
  - cytoolz==0.9.0.1=py27h14c3975_1
  - sqlite==3.24.0=h84994c4_0
  - libffi==3.2.1=hd88cf55_4
  - pandas==0.23.4=py27h04863e7_0
  - python==2.7.15=h1571d57_0
  - keras-preprocessing==1.0.2=py27_1
  - bzip2==1.0.6=h14c3975_5
  - tensorflow==1.10.0=gpu_py27h67ad7fe_0
  - pil==1.1.7=py27_2
  - mock==2.0.0=py27_0
  - pytables==3.4.4=py27ha205bf6_0
  - funcsigs==1.0.2=py27_0
  - tornado==5.1=py27h14c3975_0
  - mkl_fft==1.0.4=py27h4414c95_1
  - numexpr==2.6.8=py27hd89afb7_0
  - scipy==1.1.0=py27hfa4b5c9_1
  - openssl==1.0.2p=h14c3975_0
  - click==6.7=py27_0
  - libedit==3.1.20170329=h6b74fdf_2
  - keras-applications==1.0.4=py27_1
  - libgfortran-ng==7.3.0=hdf63c60_0
  - libgcc-ng==8.2.0=hdf63c60_1
  - freetype==2.5.5=2
  - tensorboard==1.10.0=py27hf484d3e_0
  - keras-base==2.2.2=py27_0
  - lzo==2.10=h49e0be7_2
  - snappy==1.1.7=hbae5bb6_3
  - backports_abc==0.5=py27_0
  - cupti==9.2.148=0
  - libpng==1.6.34=hb9fc6fc_0
  - msgpack-python==0.5.6=py27h6bb024c_1
  - psutil==5.4.7=py27h14c3975_0
  - cudnn==7.2.1=cuda9.2_0
  - libstdcxx-ng==8.2.0=hdf63c60_1
  - grpcio==1.12.1=py27hdbcaa40_0
  - libprotobuf==3.6.0=hdbcaa40_0

**Comment by AbdeslemSmahi** [comment]
If you are using windows , delete the build version

**Comment by Elahe96** [comment]
@AbdeslemSmahi  
Yes, I use windows, but I don't know how to do what you have said.
 Can you explain it more?

**Comment by AbdeslemSmahi** [comment]
```
name: CAL
channels:
  - pytorch
  - defaults
dependencies:
  - blas=1.0=mkl
  - ca-certificates=2019.1.23=0
  - certifi=2019.3.9=py36_0
  - cffi=1.12.3
  - cudatoolkit=10.0.130=0
  - freetype=2.9.1=ha9979f8_1
  - intel-openmp=2019.3
  - jpeg=9b
  #- libedit=3.1.20181209
  #- libffi=3.2.1=hd88cf55_4
  #- libgcc-ng=8.2.0
  #- libgfortran-ng=7.3.0=hdf63c60_0
  - libpng=1.6.37
  #- libstdcxx-ng=8.2.0=hdf63c60_1
  - libtiff=4.0.10
  - mkl=2019.3=203
  - mkl_fft=1.0.12
  - mkl_random=1.0.2
  #- ncurses=6.1=he6710b0_1
  - ninja=1.9.0=py37h74a9793_0
  - numpy-base=1.16.4
  - olefile=0.46=py36_0
  - openssl=1.1.1b
  - pip=19.1.1=py36_0
  - pycparser=2.19=py36_0
  - python=3.6.8=h9f7ef89_7
  - pytorch=1.1.0=py3.6_cuda90_cudnn7_1
  #- readline=7.0=h7b6447c_5
  - setuptools=41.0.1=py36_0
  - sqlite=3.28.0=he774522_0
  - tk=8.6.8
  - wheel=0.33.4=py36_0
  - xz=5.2.4
  - zlib=1.2.11
  - zstd=1.3.7
  - pip:
    - attrs==19.1.0
    - backcall==0.1.0
    - bcolz==1.2.1
    - bleach==3.1.0
    - cycler==0.10.0
    - decorator==4.4.0
    - defusedxml==0.6.0
    - entrypoints==0.3
    - imageio==2.5.0
    - imgaug==0.2.9
    - ipdb==0.12
    - ipykernel==5.1.1
    - ipython==7.5.0
    - ipython-genutils==0.2.0
    - ipywidgets==7.4.2
    - jedi==0.13.3
    - jinja2==2.10.1
    - joblib==0.13.2
    - jsonschema==3.0.1
    - jupyter==1.0.0
    - jupyter-client==5.2.4
    - jupyter-console==6.0.0
    - jupyter-core==4.4.0
    - kiwisolver==1.1.0
    - markupsafe==1.1.1
    - matplotlib==3.1.0
    - mistune==0.8.4
    - nbconvert==5.5.0
    - nbformat==4.4.0
    - networkx==2.3
    - notebook==5.7.8
    - numpy==1.16.4
    - opencv-python==4.1.0.25
    - pandas==0.24.2
    - pandocfilters==1.4.2
    - parso==0.4.0
    - pexpect==4.7.0
    - pickleshare==0.7.5
    - pillow==6.0.0
    - prometheus-client==0.6.0
    - prompt-toolkit==2.0.9
    - protobuf==3.8.0
    - ptyprocess==0.6.0
    - pygame==1.9.6
    - pygments==2.4.2
    - pyparsing==2.4.0
    - pyrsistent==0.15.2
    - python-dateutil==2.8.0
    - pytz==2019.1
    - pywavelets==1.0.3
    - pyzmq==18.0.1
    - qtconsole==4.5.1
    - scikit-image==0.15.0
    - scikit-learn==0.21.2
    - scipy==1.1.0
    - send2trash==1.5.0
    - shapely==1.6.4.post2
    - six==1.12.0
    - terminado==0.8.2
    - testpath==0.4.2
    - torch==1.1.0
    - torchvision==0.2.2.post3
    - tornado==6.0.2
    - tqdm==4.32.1
    - traitlets==4.3.2
    - wcwidth==0.1.7
    - webencodings==0.5.1
    - widgetsnbextension==3.4.2
prefix: /home/rsi/anaconda3/envs/CAL
```
you can use this , i think it works for windows

**Comment by Marwa215** [comment]
Hi @AbdeslemSmahi, have you trained the model, because I have trained it but it gives no predictions. If you trained it could I get the trained model. 
Thanks

**Comment by Elahe96** [comment]
@AbdeslemSmahi 
Thank you. 
I have tried what you said, and I figured out some other packages also should be omitted, but as some packages are not available with the described version in windows, I faced problems running the training code. Finally, I installed ubuntu beside windows as dual boot format, and all problems have been solved.

**Comment by Marwa215** [comment]
Hi @Elahe96, did you get right predictions after training the pytorch model?!

**Comment by Elahe96** [comment]
Hi @Marwa215 , I had the issue mentioned in (Unable to find gru.pth model  #18), and after training, the problem has been solved, and now I can run the agent correctly.

**Comment by Marwa215** [comment]
Hi @Elahe96, thanks for your reply. I have trained the model many times with different feature extractors such as ResNet50, ResNet34 but the predictions always give the default values e.g. the traffic-light=False although it is red in the current scene.
Could you share your training code and the steps you followed for training. Or if you are not mind to send me the trained model 

Thanks

**Comment by Elahe96** [comment]
Hi @Marwa215 
I have used the same training code as provided here on the training folder. I didn't uncomment the feature extractor part to be trained. I have used "python python_client/driving_benchmark.py -c Town01 -v -n test --corl-2017" for testing the trained code in Carla. Where should I check the predictions?
The predictions that are shown on the console doesn't contain the traffic light. For the predictions that are shown on the console, the result is correct, but the traffic light is not considered as it is not available on the predictions that I can see.
These are the things that I can see on the console:

INFO: Time for prediction: 0.05220341682434082
INFO: CARLA Direction 5.0, Real Direction 0
INFO: STATE: cruising, PROBA: 100.0000 %
INFO: DECAY: 0.09992162685955484
INFO: Controller is Inputting:
INFO: Steer = 0.000000 Throttle = 0.664728 Brake = 0.000000 
INFO: Distance to target: 72.366771
INFO: offroad: False, otherlane: False, coll_other: 0.0, old_coll: 0
INFO: Time remaining: 2 m 51 s

**Comment by Marwa215** [comment]
Hi @Elahe96, so how do you know that the model it trained well?  I compared the training loss and validation loss during the training and the results are not satisfying. I have printed the predictions values within this results from Call_agent script but all predictions give the default values such as speed limit if not 30 or 60 or 90 it print -1 and the predicted value is always-1 even if the speed limit changed in the simulator. Another way to check that the agent should takes into consideration these predictions but this almost depends on the adjusted agent not the trained model.  Does the agent stopped at red traffic light or changed speed with the new speed limit?

**Comment by Elahe96** [comment]
Hi @Marwa215
The best validation loss, as printed in the notebook, is 0.060737. The printed predictions are changed while running the client; for example, the offroad and otherlane get different values based on what is shown in the simulator. But the speed limit and the traffic lights are not considered as they are not parts of the predictions which are printed on the console. I had taken a look at results printer, and affordances are not available there. I thought maybe it is because of the benchmark that these affordances are not considered...
How did you check the prediction results?
I don't get any result about affordance on the console while running driving_benchmark.py ....

**Comment by Marwa215** [comment]
Hi @Elahe96, changing lane and off-road are measurements within Carla not predicted values through the model. Read the authors paper you can find only 6 predictions traffic light, speed limit, hazard prediction, distance and angle from road center and distance from leading vehicle. If the current running agent not respect these 6 predictions then the model didn’t trained well or the agent not adjusted well to consider these predictions.

**Comment by zzzwww** [comment]
Hi @Elahe96 
There is a problem when I use "python python_client/driving_benchmark.py -c Town01 -v -n test --corl-2017", the result shows "unable to find gru.pth model". I wonder how you solved the problem. Do I need to train the model? And I didn't find the dataset, either. Do I need to get dataset by myself? Looking forward to your answer.

**Comment by Elahe96** [comment]
Hi @zzzwww
Yes, you need to train the model, as mentioned in #18. The training data is addressed at the end of the main page https://github.com/xl-sr/CAL. 
#### to download and untar the dataset, you should run:
wget https://s3.eu-central-1.amazonaws.com/avg-projects/conditional_affordance_learning/dataset.tar.gz
tar -xzvf dataset.tar.gz

**Comment by zzzwww** [comment]
Hi @Elahe96 
Thank you very much. But I can't download the dataset by running the code wget https://s3.eu-central-1.amazonaws.com/avg-projects/conditional_affordance_learning/dataset.tar.gz, perhaps the site has been deactivated.

**Comment by zzzwww** [comment]
Hi @Elahe96 @Marwa215 
I have tried to download the dataset via "wget https://s3.eu-central-1.amazonaws.com/avg-projects/conditional_affordance_learning/dataset.tar.gz", but failed. It seems that the link is invalid. Do you still have the dataset? Could you please send it to me? My email adress is 3478629045@qq.com. Looking forward to your reply.

**Comment by Elahe96** [comment]
Hi @zzzwww,
I have checked the link, and it's working if you can't download it, try downloading it by copying the link https://s3.eu-central-1.amazonaws.com/avg-projects/conditional_affordance_learning/dataset.tar.gz to your browser address. The data is too big, and I can't email it, sorry.
